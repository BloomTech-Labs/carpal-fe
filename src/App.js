import React from "react";
import { Route } from "react-router-dom";
import TopNav from "./Components/Nav/TopNav";
import Nav from "./Components/Nav/Nav";
import Login from "./Components/Login-SignUp/Login";
import SignUp from "./Components/Login-SignUp/SignUp";
import Dashboard from "./Components/Dashboards/Dashboard";
import ProfilePage from "./Components/Profile-Pages/Profile-Pages";
import LandingPage from "./Components/Landing-Page/Landing-Page";
import ProtectedRoute from "./Utils/ProtectedRoute";
import RideFind from "./Components/Rides/RideFind/RideFind";
import Logout from "./Components/Login-SignUp/Logout";
import SavedRide from "./Components/Rides/SavedRide/SavedRide";
import RideRequests from "./Components/Rides/RideRequests/RideRequests";

import "./App.scss";
import profileImage from "./Components/Profile-Pages/Profile-Image/Profile-img-upload";

//TODO: update app to include loader if isLoading

function App() {
    return (
        <div className="App" role="App">
            <TopNav />
            <div className="container">
                <Route exact path="/upload" component={profileImage} />
                <Route exact path="/login" component={Login} />
                <Route exact path="/signup" component={SignUp} />
                <ProtectedRoute exact path="/logout" component={Logout} />
                <ProtectedRoute
                    exact
                    path="/profilepage"
                    component={ProfilePage}
                />
                <Route exact path="/saved" component={SavedRide} />

                <ProtectedRoute exact path="/dashboard" component={Dashboard} />
                <Route exact path="/" component={LandingPage} />
                <ProtectedRoute exact path="/Home" component={RideFind} />
                <ProtectedRoute
                    exact
                    path="/requests"
                    component={RideRequests}
                />
            </div>
            <Nav />
        </div>
    );
}
export default App;
