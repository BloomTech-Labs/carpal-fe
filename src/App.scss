.container {
    max-width: 1500px;
    margin: 0 auto;
    padding: 70px 0;
}

input:focus{
    outline: none;
}

button:focus{
    outline: none;
}