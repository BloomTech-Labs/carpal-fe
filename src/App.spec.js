import React from "react";
import App from "./App";
import { BrowserRouter as Router } from "react-router-dom";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(rtl.cleanup);

describe("renders without crashing", () => {
    test("displays CarPal", () => {
        const wrapper = rtl.render(
            <Router>
                <App />
            </Router>
        );

        const title = wrapper.queryByRole(/app/i);
        expect(title).toBeInTheDocument();
        expect(title).toBeVisible();
    });
    describe("creates a snapshot and render", () => {
        test("render snapshot", () => {
            const tree = rtl.render(
                <Router>
                    <App />
                </Router>
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
