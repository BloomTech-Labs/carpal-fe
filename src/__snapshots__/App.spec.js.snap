// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing creates a snapshot and render render snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
        role="App"
      >
        <div
          class="top-nav"
        >
          <a
            href="/profilepage"
          >
            <img
              alt="Patchy"
              class="profilePic"
              src="Patchyv2.0.png"
            />
          </a>
          <a
            href="/"
          >
            CarPal
          </a>
          <nav
            class="hamburger-nav"
            id="hamburger-nav"
          >
            <a
              href="/login"
            >
              Login
            </a>
            <a
              href="/signup"
            >
              Sign Up
            </a>
          </nav>
          <button
            class="hamburger"
            id="hamburger"
          >
            <div
              class="navicon bar1"
            />
            <div
              class="navicon bar2"
            />
            <div
              class="navicon bar3"
            />
            <div
              class="hidden-text"
            >
              Menu
            </div>
          </button>
        </div>
        <div
          class="container"
        >
          <div
            class="mainContainer"
          >
            <div
              class="hero"
            >
              <video
                autoplay=""
                class="slider"
                loop=""
              >
                <source
                  src="CarPal.mp4"
                  type="video/mp4"
                />
              </video>
              <a
                href="/Signup"
              >
                <p
                  class="overlaybutton"
                >
                  Sign Up
                </p>
              </a>
            </div>
            <div
              class="main"
            >
              <div
                class="intro"
              >
                <h1
                  class="introTitle"
                  data-testid="welcomeh1"
                >
                  Welcome to CarPal
                </h1>
                <p
                  class="firstpara"
                  data-testid="firstptag"
                >
                  The future of networking and commuting at the same time. Yeah, we did it seriously. Watch this, with Carpal you will be able to join a community of drivers and riders who are looking to save money, lessen the impact on the enviorment, and gain a friend along the way.
                </p>
              </div>
              <section
                class="container"
              >
                <div
                  id="first"
                >
                  <h2
                    data-testid="h2tag1"
                  >
                    Features
                  </h2>
                  <div
                    class="features"
                  >
                    <div
                      class="paraone"
                    >
                      <p
                        class="para"
                      >
                        With this awesome app, these are just a few of the features you will get such as SMS messaging, friends list, gamification (user stats and point system) and even driver's real-time location, but wait there's more and its just five clicks away front becoming a Carpaller.
                      </p>
                    </div>
                    <div
                      class="imgone"
                    >
                      <img
                        alt="Temp img"
                        class="allimg"
                        role="image"
                        src="Group 38 (1).png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  id="second"
                >
                  <h2
                    data-testid="h2tag2"
                  >
                    Drivers
                  </h2>
                  <div
                    class="featuresflip"
                  >
                    <div
                      class="paraone"
                    >
                      <p
                        class="para"
                      >
                        Are you a current driver who loves to drive or even hates to drive, well guess what you in luck. The Carpal developer kept you in mind when building this app I might even add that they made it just for you.
                      </p>
                    </div>
                    <div
                      class="imgone"
                    >
                      <img
                        alt="Temp img"
                        class="allimg"
                        src="Group 29.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  id="third"
                >
                  <h2
                    data-testid="h2tag3"
                  >
                    Riders
                  </h2>
                  <div
                    class="features"
                  >
                    <div
                      class="paraone"
                    >
                      <p
                        class="para"
                      >
                        As a rider you gain many great features and here are just a few of those. Request a ride to your destination, customize the type of driver you are interested in meet, and rate each Carpal experience to make sure our community stays the best in the business.
                      </p>
                    </div>
                    <div
                      class="imgone"
                    >
                      <img
                        alt="Temp img"
                        class="allimg"
                        src="Group 39.png"
                      />
                    </div>
                  </div>
                </div>
              </section>
            </div>
            <section
              class="containertwo"
            >
              <h2
                class="padding"
              >
                Ready to gooooooooo?
              </h2>
              <a
                href="/Signup"
              >
                <button
                  name="signup"
                >
                  sign Up
                </button>
              </a>
            </section>
          </div>
        </div>
        <nav
          class="bottom-nav"
        >
          <a
            class="fas fa-home"
            href="/dashboard"
          />
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
      role="App"
    >
      <div
        class="top-nav"
      >
        <a
          href="/profilepage"
        >
          <img
            alt="Patchy"
            class="profilePic"
            src="Patchyv2.0.png"
          />
        </a>
        <a
          href="/"
        >
          CarPal
        </a>
        <nav
          class="hamburger-nav"
          id="hamburger-nav"
        >
          <a
            href="/login"
          >
            Login
          </a>
          <a
            href="/signup"
          >
            Sign Up
          </a>
        </nav>
        <button
          class="hamburger"
          id="hamburger"
        >
          <div
            class="navicon bar1"
          />
          <div
            class="navicon bar2"
          />
          <div
            class="navicon bar3"
          />
          <div
            class="hidden-text"
          >
            Menu
          </div>
        </button>
      </div>
      <div
        class="container"
      >
        <div
          class="mainContainer"
        >
          <div
            class="hero"
          >
            <video
              autoplay=""
              class="slider"
              loop=""
            >
              <source
                src="CarPal.mp4"
                type="video/mp4"
              />
            </video>
            <a
              href="/Signup"
            >
              <p
                class="overlaybutton"
              >
                Sign Up
              </p>
            </a>
          </div>
          <div
            class="main"
          >
            <div
              class="intro"
            >
              <h1
                class="introTitle"
                data-testid="welcomeh1"
              >
                Welcome to CarPal
              </h1>
              <p
                class="firstpara"
                data-testid="firstptag"
              >
                The future of networking and commuting at the same time. Yeah, we did it seriously. Watch this, with Carpal you will be able to join a community of drivers and riders who are looking to save money, lessen the impact on the enviorment, and gain a friend along the way.
              </p>
            </div>
            <section
              class="container"
            >
              <div
                id="first"
              >
                <h2
                  data-testid="h2tag1"
                >
                  Features
                </h2>
                <div
                  class="features"
                >
                  <div
                    class="paraone"
                  >
                    <p
                      class="para"
                    >
                      With this awesome app, these are just a few of the features you will get such as SMS messaging, friends list, gamification (user stats and point system) and even driver's real-time location, but wait there's more and its just five clicks away front becoming a Carpaller.
                    </p>
                  </div>
                  <div
                    class="imgone"
                  >
                    <img
                      alt="Temp img"
                      class="allimg"
                      role="image"
                      src="Group 38 (1).png"
                    />
                  </div>
                </div>
              </div>
              <div
                id="second"
              >
                <h2
                  data-testid="h2tag2"
                >
                  Drivers
                </h2>
                <div
                  class="featuresflip"
                >
                  <div
                    class="paraone"
                  >
                    <p
                      class="para"
                    >
                      Are you a current driver who loves to drive or even hates to drive, well guess what you in luck. The Carpal developer kept you in mind when building this app I might even add that they made it just for you.
                    </p>
                  </div>
                  <div
                    class="imgone"
                  >
                    <img
                      alt="Temp img"
                      class="allimg"
                      src="Group 29.png"
                    />
                  </div>
                </div>
              </div>
              <div
                id="third"
              >
                <h2
                  data-testid="h2tag3"
                >
                  Riders
                </h2>
                <div
                  class="features"
                >
                  <div
                    class="paraone"
                  >
                    <p
                      class="para"
                    >
                      As a rider you gain many great features and here are just a few of those. Request a ride to your destination, customize the type of driver you are interested in meet, and rate each Carpal experience to make sure our community stays the best in the business.
                    </p>
                  </div>
                  <div
                    class="imgone"
                  >
                    <img
                      alt="Temp img"
                      class="allimg"
                      src="Group 39.png"
                    />
                  </div>
                </div>
              </div>
            </section>
          </div>
          <section
            class="containertwo"
          >
            <h2
              class="padding"
            >
              Ready to gooooooooo?
            </h2>
            <a
              href="/Signup"
            >
              <button
                name="signup"
              >
                sign Up
              </button>
            </a>
          </section>
        </div>
      </div>
      <nav
        class="bottom-nav"
      >
        <a
          class="fas fa-home"
          href="/dashboard"
        />
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
